pkg/webhook/preflight/generic/spec.go

```diff
@@ -45,7 +45,8 @@ func newConfigurationCheck(
 		configurationCheck.result.InternalError = true
 		configurationCheck.result.Causes = append(configurationCheck.result.Causes,
 			preflight.Cause{
-				Message: fmt.Sprintf("Failed to unmarshal cluster variable %s: %s",
+				Message: fmt.Sprintf(
+					"Failed to unmarshal cluster variable %q: %s. This error should not happen under normal circumstances. Please report it.", ///nolint:lll // Message is long.
```

Comment:
my 2 cents: I think we should not explicitly mention "Please report it" in generic messages. Comparing it to 400 (bad requests) or 500 (internal error) . some of this is retryable, some needs inspection and some might need external support. Also we dont provide more context about reporting too.

Reply:
Addressed in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1210/commits/4023398b85ab2738872d2c96bf66546dcfaa82f0 and https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1210/commits/4e94afe67e72402596f6528b76df211d8a47d069

Reply:
Thank you.

Reply:
Note that, when a preflight check panics, the message asks the user to report the stack trace, e.g.

```
The preflight check code had a specific internal error called a \"panic\". This error should not happen under normal circumstances. Please report it, and include the following information: <stacktrace ...>.
```